### PR TITLE
JP-3005: Update core schema for new step status keyword in metadata

### PR DESCRIPTION
**Pull request by tapastro**
Resolves [JP-3005](https://jira.stsci.edu/browse/JP-3005)
Resolves [RCAL-nnnn](https://jira.stsci.edu/browse/RCAL-nnnn)

Closes #

This PR adds the cal_step keyword to denote whether pixel replacement was run, skipped, etc. It appears that I don't have many permissions here, as I can't add milestones/labels or request reviewers.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [x] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml
Comment (hbushouse) [inline src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml:2405]: not a big deal, but could go ahead and just spell out the full name to `pixel_replacement`

**Review by hbushouse — APPROVED** [review]
LGTM

File: src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml
Comment (hbushouse) [inline src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml:2405]: Ack. I see from https://github.com/spacetelescope/jwst/pull/7398/files that the actual name of the step is in fact "pixel_replace" (not replace*ment*), so I take back my previous suggestion and this should be changed back to just "pixel_replace". Sorry about the hassle.

Comment (tapastro) [reply src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml]: No worries, flipped it back.

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
> @tapastro and @hbushouse It looks like the failure in the jwst CI job is expected (due to the changes in this PR): https://github.com/spacetelescope/stdatamodels/actions/runs/4622302983/jobs/8174828584?pr=149#step:10:192
> 
> However the failing jwst job is preventing this from being merged because of the branch protections. Any objection to removing the jwst job passing as a requirement for PRs in this package?

I would not permanently disable the jwst job as a requirement for passing. I/we (those of us with appropriate privilege) can just override that requirement on a case-by-case basis when deemed appropriate (like this one). I just check the magic box that says "Merge without waiting for requirements to be met" and we're good to go.

**Comment by hbushouse** [comment]
Oh no, that would be WAY too dangerous!  ;-)

**Comment by braingram** [comment]
I was in the mood for some Kenny Loggins :)

@tapastro are you now able to use the 'merge without waiting for requirements' option. I added you with 'write' permissions to this repo. Looking over the github docs I don't think this is high enough permissions but figured I'd check. If you're not able to see the options appear to be:
- add you to the stdatamodels-maintainers team, which will mean you receive notifications due to your inclusion in the CODEOWNERS file.
- add you as an admin to this repository, which will give you additional permissions beyond allowing for bypassing branch protections
- specifically add you as allowed to bypass branch protections

I'm also happy to just merge this PR if the added permissions seem too complicated for how often you expect to make these types of changes. I'm happy with any of the above options, the first seems reasonable to me if you expect to make additional changes to stdatamodels.

**Comment by tapastro** [comment]
> @tapastro are you now able to use the 'merge without waiting for requirements' option?

No, though it does look like I could merge if the requirements passed. Thanks!

**Comment by braingram** [comment]
> No, though it does look like I could merge if the requirements passed. Thanks!

Thanks for checking. @tapastro I added you to the stdatamodels-maintainer team. Mind giving it one more try?

**Comment by hbushouse** [comment]
I would like to hold off just a bit before merging this, because as soon as we do, anyone using stdatamodels from github/master may run into trouble until necessary updates in the jwst package have also been merged. Let's wait till the PR over in JWST is ready and then we'll merge them both at nearly the same time.

**Comment by tapastro** [comment]
> > No, though it does look like I could merge if the requirements passed. Thanks!
> 
> Thanks for checking. @tapastro I added you to the stdatamodels-maintainer team. Mind giving it one more try?

I do appear to have unnatural powers now. 😟

**Pull request merged**
